### PR TITLE
feature/add-matter-status-constants

### PR DESCRIPTION
**Pull request by tohuynh**
**Pull request recommendations:**
- [x] Name your pull request _your-development-type/short-description_. Ex: _feature/read-tiff-files_
- [ ] Link to any relevant issue in the PR description. Ex: _Resolves [gh-12], adds tiff file format support_
- [x] Provide context of changes.
- Add useful matter status constants for `MatterStatus` database model. The constants are `Adopted`, `Rejected`, and `In Progress`.
- Provide validators for them in the database models. Instead of creating a validator for each constant class, I created a factory function that creates the validator to avoid duplication.
- Also added `status` to the `Matter` ingestion model. I decided to put it there to follow the `sponsors` example. Similar to how `sponsors` will be used to create/update `MatterSponsor`, `status` will be used to create/update `MatterStatus` in the pipeline.
- [x] Provide relevant tests for your feature or bug fix.
- [ ] Provide or update documentation for any feature added by your pull request.

Thanks for contributing!

### CODE REVIEW

File: cdp_backend/pipeline/ingestion_models.py [hunk 1]
```diff
@@ -97,6 +97,7 @@ class Matter:
     name: str
     matter_type: str
     title: str
+    status: str
```
Comment (JacksonMaxfield) [inline cdp_backend/pipeline/ingestion_models.py:100]: Can we change this to `result_status` to clarify that this is the status of the matter _after_ the event.

**Review by JacksonMaxfield — APPROVED** [review]
This makes a whole lot of sense to me. Approved but can we change the name from `status` to `result_status` in the ingestion model for clarity.

**Comment by tohuynh** [comment]
@isaacna Just a heads up, there's an additional field `result_status` (the matter's status after the event) in the `Matter` ingestion model. It will be used in the pipeline to update the `MatterStatus` db collection.

**Comment by isaacna** [comment]
> @isaacna Just a heads up, there's an additional field `result_status` (the matter's status after the event) in the `Matter` ingestion model. It will be used in the pipeline to update the `MatterStatus` db collection.

Sounds good thanks for letting me know

**Pull request merged**